packages/odata-service-writer/src/data/annotations.ts

```diff
@@ -0,0 +1,101 @@
+import { XMLParser } from 'fast-xml-parser';
+import { t } from '../i18n';
+import { NamespaceAlias, OdataService } from '../types';
+
+/**
+ * Updates the passed odata service with the namespaces parsed from the specified metadata and annotations.
+ *
+ * @param {Partial<OdataService>} service - an odata service where at least metadata and annotations properties are defined
+ * @param {string} service.metadata - OData service metadata xml
+ * @param {string} service.annotations - OData service annotations xml
+ * @returns A reference to the namspaces array
+ */
+export function getAnnotationNamespaces({ metadata, annotations }: Partial<OdataService>): NamespaceAlias[] {
+    // enhance service with annotations namespaces
+    const schemaNamespaces = metadata ? getNamespaces(metadata) : [];
+
+    return schemaNamespaces.map((schema: NamespaceAlias) => {
+        // Check if alias exists in backend annotation file, if so use it
+        if (annotations) {
+            const annotationAlias =
+                annotations.xml && schema.namespace
+                    ? getAliasFromAnnotation(annotations.xml, schema.namespace)
```

Comment:
Does it make sense to memoize this function call or store an intermediate result? I'm not sure how performant XML parsing is.

Reply:
Memoize pattern is generally useful when the same combination arguments are passed multiple times, which in the case of `getAliasFromAnnotation` wont happen (unless there are duplicated `namespace` references). However, it would might be beneficial to only parse the same xmls once for sure. Im guessing thats what you are referring to. 

In the case of multiple namespaces the same `annotation` xml will be parsed repeatedly. We would need to generate an id for each passed xml as an equality check and store this along with the parsed result. Im not sure its worth the effort currently as `fast-xml-parser` is quite performant. @devinea @tobiasqueck any thoughts?

Reply:
Alternatively we could parse the `annotation` xml before the call to `getAliasFromAnnotation` and pass the parsed result. This would save some processing and is a minor update, I think this makes sense.